src/analysis/GameAnalyzer.cpp

```diff
@@ -145,12 +145,12 @@ bool GameAnalyzer::isClosingInToGoal(const Field &field, RobotPtr robot, bool ou
     return false;
 }
 
-void GameAnalyzer::start(int iterationsPerSecond) {
+void GameAnalyzer::start(world::Field const & field, int iterationsPerSecond) {
     if (!running && world::world->weHaveRobots()) {
         std::cout << "GameAnalyzer: "
                   << "Starting at " << iterationsPerSecond << " iterations per second" << std::endl;
         auto delay = (unsigned)(1000.0 / iterationsPerSecond);
-        thread = std::thread(&GameAnalyzer::loop, this, delay);
+        thread = std::thread(&GameAnalyzer::loop, this, field, delay);
```

Comment:
This should give problems since the const & field will not change after GameAnalyzer is started. I feel like a pointer/something else should work better.

Reply:
As in, if the field changes after GameAnalyzer is started we have a problem.

Reply:
It is already given a const ref to field as params in the start function, so I think this should work